File: src/modifiers/accessibilityElement.ts
import { createModifier } from './createModifier';

/// How a SwiftUI accessibility element treats its children.
/// - `'ignore'` — children are not accessible (the element is a leaf)
/// - `'combine'` — children are merged into this element
/// - `'contain'` — children remain individually accessible inside it
export type AccessibilityChildBehavior = 'ignore' | 'combine' | 'contain';

/// SwiftUI `.accessibilityElement(children:)`. Creates an accessibility
/// element for the view, controlling how its children participate.
/// Defaults to `'ignore'`, matching SwiftUI.
export function accessibilityElement(
  children: AccessibilityChildBehavior = 'ignore'
) {
  return createModifier('accessibilityElement', { children });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityElement.swift
import SwiftUI

/// SwiftUI `.accessibilityElement(children:)`. Creates an accessibility
/// element for the view, controlling how its children participate.
/// `children` maps to `AccessibilityChildBehavior`; defaults to `.ignore`.
enum RNWAccessibilityElementModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityElement") { view, params, _ in
            let behavior: AccessibilityChildBehavior
            switch params.string("children") {
            case "combine": behavior = .combine
            case "contain": behavior = .contain
            default:        behavior = .ignore
            }
            return AnyView(view.accessibilityElement(children: behavior))
        }
    }
}
